std/path.d

```diff
@@ -3269,7 +3269,7 @@ body
                             /* Match for:
                              *   pattern[pi0..pi-1] ~ pattern[piRemain..$]
                              */
-                            if (pattmp.ptr == null)
+                            if (() @trusted { return pattmp.ptr; }() == null)
```

Comment:
oh I see what you mean. The compiler generates some rather stupid code for `array is null`. Should be a topic for another issue.